Competition2018/src/main/java/competition/subsystems/elevator/commands/ElevatorMaintainerCommand.java

```diff
@@ -24,16 +39,36 @@ public void initialize() {
         log.info("Initializing");
         if (!elevator.isCalibrated()) {
             log.warn("ELEVATOR UNCALIBRATED - THIS COMMAND WILL NOT DO ANYTHING!");
+            startTime = Timer.getFPGATimestamp();
+            attemptTime = startTime + calibrationAttemptTime.get();
         }
     }
 
     @Override
     public void execute() {
+        boolean maintain = false;
+        boolean manual = false;
+        boolean tryToCalibrate = false;
         if (elevator.isCalibrated()) {
-            double power = pid.calculate(elevator.getTargetHeight(), elevator.getCurrentHeightInInches());
-            elevator.setPower(power);
-        } else {
-            elevator.stop();
+            maintain = true;
         }
-    }
+        else if (Timer.getFPGATimestamp() < attemptTime) {
+            tryToCalibrate = true;
+        }
+        else {
+            manual = true;
+        }
+        
+        
+        
+        if (maintain) {
+            elevator.setPower(pid.calculate(elevator.getTargetHeight(), elevator.getCurrentHeightInInches()));
+        }
+        else if (tryToCalibrate) {
+            elevator.setPower(-1);
```

Comment:
This should be a property

Reply:
what specifically?

Reply:
@WasabiFan I don't think it's currently relevant - the Elevator subsystem already has power restrictions if it's uncalibrated, which are controlled by the DoubleProperty `calibrationPower`

Reply:
I don't think that is the right solution. Relying on an assumed internal behavior of the subsystem for this command's functionality isn't safe and confuses a reader. If the subsystem universally limits power while it's uncalibrated, that should be called `Uncalibrated max power` and this command should have a separate property for the speed it uses to calibrate.

Reply:
changed "elevator.setPower(-1)" to "elevator.lower()"